client/src/Hooks/useReactionsFetch.js

```diff
@@ -1,20 +1,27 @@
 import { useState, useEffect } from 'react';
 import api from '../api.js';
-export default function useReactionsFetch (username) {
+export default function useReactionsFetch (username, setPopup) {
 	const [reactions, setReactions] = useState({});
-	const [error, setError] = useState();
 	const [loading, setLoading] = useState(true);
 	useEffect(() => {
-		api.sendReactionsRequest(username).then((res) => {
-			setReactions(res.data);
-			setLoading(false);
-		}).catch((error) => {
-			setError(error);
-		});
+		async function fetchData () {
+			if (!username) {
+				setLoading(false);
+				return;
+			}
+			try {
+				const res = await api.sendReactionsRequest(username);
+				if (res.status === 200) {
+					setReactions(res.data);
+				}
+				setLoading(false);
+			} catch (e) {
+				setPopup('Something went wrong when fetching reactions');
+			}
```

Comment:
The similar question: what if an error happens, what would be with loading state?

Reply:
if there is an error I don't want to set the state to false as then the UI will try to display the reactions when there is none

Reply:
but I see the issues, loading in this case should be false since the request is done